### PR TITLE
Add support for untyped scripts in `tasty-plutus`

### LINKED ISSUE #198: Export the WrappedValidator constructor
Sam was mightily annoyed at the mob coding session yesterday that this constructor wasn't exported from `Test.Tasty.Plutus.TestScript`. That leaves `toTestValidator` as the only way to construct a `WrappedValidator`, but this function expects a typed validator. It can't be used to wrap an untyped Plutus validator that errors out instead of returning `False`.

### PR DESCRIPTION
**Pull request by re-xyr**
Fixes #198.

- Allows the construction of `TestScript`s from untyped validators
- Internally, uses `error` to signal script failure (including returning `False`)

@blamario Does this look good to you?

### CODE REVIEW

File: tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs [hunk 1]
```diff
@@ -14,45 +14,44 @@ import Plutus.V1.Ledger.Api (
...
-  ScriptError,
+  ScriptError (EvaluationError, EvaluationException, MalformedScript),
   runMintingPolicyScript,
   runScript,
  )
 import Safe (lastMay)
 
--- The result of parsing a log from a script emulation
+-- The result of parsing a log from a errored script emulation
 data ScriptResult
```
Comment (blamario) [inline tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs:24]: `ScriptResult` now contains only failing alternatives, so it should probably be renamed to `ScriptFailure`.

Comment (blamario) [reply tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs:24]: On the second thought, do we really need a new data type at all? What's wrong with just returning the same `ScriptError` we got from Plutus?

File: tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs [hunk 2]
```diff
@@ -14,45 +14,44 @@ import Plutus.V1.Ledger.Api (
...
-    Nothing -> InternalError t
-    Just t' -> ParseFailed t'
+parseError :: ScriptError -> ScriptResult
+parseError = \case
+  EvaluationError logs msg ->
+    case lastMay logs >>= Text.stripPrefix "tasty-plutus: Parse failed: " of
+      Nothing -> ScriptFailed logs msg
+      Just t -> ParseFailed logs t
+  EvaluationException name msg -> PlutusEvaluationException name msg
+  MalformedScript msg -> PlutusMalformedScript msg
```
Comment (blamario) [inline tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs:57]: The `MalformedScript` constructor is gone from the latest Plutus, you may want to disregard this case.

File: tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs (hunk 1 above)
Comment (re-xyr) [reply tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs:24]: `ScriptResult` has separate cases for intentional script error and `tasty-plutus` parse error, that's why I didn't use `ScriptError` directly. I'd love to switch that to a better formulation if you have any suggestions.

File: tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs (hunk 2 above)
Comment (blamario) [reply tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs:57]: If you do, you'll have a problem finding a compatible version of `plutus-apps`. It's lagging behind `plutus`. I had to use this fork a couple weeks ago: https://github.com/mlabs-haskell/plutus-apps/tree/mario/MalformedScript

Comment (kozross) [reply tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs:57]: @blamario From what I could determine, [this commit](https://github.com/input-output-hk/plutus/commit/4127e9cd6e889824d724c30eae55033cb50cbf3e) is what `plutus-apps` is pinned to. It's only 2 weeks old - are the changes you need newer than this, or am I missing something here?

Comment (blamario) [reply tasty-plutus/src/Test/Tasty/Plutus/Internal/Run.hs:57]: No the particular change to `plutus` was older than this. Try your luck with the latest `plutus-apps` then.

**Review by kozross — APPROVED** [review]

**Pull request merged**